kmz_to_kml.py

```
@@ -0,0 +1,207 @@
+
+    def parse_kml(self):
+        parse_list = []
+        registration_id = self.get_registration()
+        with open(FileNames.KML.value, 'r') as kml:
+            parse = BeautifulSoup(kml, 'lxml-xml')
+
+        for placemark in parse.find_all('Placemark'):
+            name = placemark.find('name')
+            if name is None:
+                self.integration_log.add_log(LogLevel.ERROR.log_level_name, f'Failed parse kml. Exception [Field name not found]')
+                raise Exception(f'Failed parse kml. Exception [Field name not found]')
+
+            name = name.text
+
+            description = placemark.find('description')
+            if description is None:
+                self.integration_log.add_log(LogLevel.ERROR.log_level_name, f'Failed parse kml. Exception [Field description not found]')
+                raise Exception(f'Failed parse kml. Exception [Field description not found]')
+
+            description = description.text.strip()
+
+            for split in re.split('<br/>', description):
+                if re.search('Fire Type', split) is None:
+                    continue
+                else:
+                    fire_type = re.split('</b>', split)[1].strip()
+                    break
+
+            if fire_type is None:
+                self.integration_log.add_log(LogLevel.ERROR.log_level_name, f'Failed parse kml. Exception [Field fire_type not found]')
+                raise Exception(f'Failed parse kml. Exception [Field fire_type not found]')
+
+            try:
+                fire_size = re.search(r'\d+', re.search(r'\d+\sacres|\d+acres', description).group()).group()
+            except Exception:
+                self.integration_log.add_log(LogLevel.ERROR.log_level_name, f'Failed parse kml. Exception [Field fire_size not found]')
+                raise Exception(f'Failed parse kml. Exception [Field fire_size not found]')
```

Lost original exception

---

The original error would say that you cannot call the group function for an empty value, which means that no fire_size value was found.
If it is necessary, I can split this line and add some conditions.

---

you know all exceptions which can be thrown? Never swallow original exception

---

Got it, fixed it

---

Oh, it's OK. Sorry. 
"Exception chaining happens automatically when an exception is raised inside an except or finally section."

---

you have to **handle**, **not collect** exceptions in a single place

---

I have already handled these exceptions by adding conditions